packages/framework/src/Model/AdvancedSearch/ProductAdvancedSearchConfig.php

```
@@ -40,10 +37,19 @@ public function __construct(
         $this->registerFilter($productNameFilter);
         $this->registerFilter($productCatnumFilter);
         $this->registerFilter($productPartnoFilter);
-        $this->registerFilter($productStockFilter);
         $this->registerFilter($productFlagFilter);
         $this->registerFilter($productCalculatedSellingDeniedFilter);
         $this->registerFilter($productBrandFilter);
         $this->registerFilter($productCategoryFilter);
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\AdvancedSearchFilterInterface $filter
+     */
+    protected function unregisterFilter(AdvancedSearchFilterInterface $filter): void
+    {
+        if (array_key_exists($filter->getName(), $this->filters)) {
```

Maybe throw an exception if the filter is not registered?